cpp/include/raft/sparse/linalg/laplacian.cuh

```diff
@@ -20,8 +20,8 @@
 #include <type_traits>
 
 namespace raft {
-namespace spectral {
-namespace matrix {
+namespace sparse {
+namespace linalg {
```

Comment:
Sorry, I just realized this was detail code. Can we please pull this into the `detail/` directory and expose a wrapper function around it? That's how we hide the impl details (since this is header only) to make it blatantly clear to users where the public APIs are (it's been more challenging to do this than it should be for header-only).